Content.Server/Chemistry/EntitySystems/SolutionInjectOnCollideSystem.cs

```diff
@@ -2,24 +2,40 @@
 using Content.Server.Body.Systems;
 using Content.Server.Chemistry.Components;
 using Content.Server.Chemistry.Components.SolutionManager;
+using Content.Server.Chemistry.Events;
+using Content.Server.Nutrition.EntitySystems;
+using Content.Server.Popups;
+using Content.Shared.Inventory;
 using JetBrains.Annotations;
-using Robust.Shared.GameObjects;
-using Robust.Shared.IoC;
 using Robust.Shared.Physics.Dynamics;
 
+
 namespace Content.Server.Chemistry.EntitySystems
 {
     [UsedImplicitly]
     internal sealed class SolutionInjectOnCollideSystem : EntitySystem
     {
         [Dependency] private readonly SolutionContainerSystem _solutionsSystem = default!;
         [Dependency] private readonly BloodstreamSystem _bloodstreamSystem = default!;
+        [Dependency] private readonly InventorySystem _inventorySystem = default!;
 
         public override void Initialize()
         {
             base.Initialize();
             SubscribeLocalEvent<SolutionInjectOnCollideComponent, ComponentInit>(HandleInit);
             SubscribeLocalEvent<SolutionInjectOnCollideComponent, StartCollideEvent>(HandleInjection);
+            SubscribeLocalEvent<SolutionInjectOnCollideComponent, TransferThroughFaceAttemptEvent>(OnTransferThroughFaceAttemptEvent);
+        }
+
+        private void OnTransferThroughFaceAttemptEvent(EntityUid uid, SolutionInjectOnCollideComponent component , TransferThroughFaceAttemptEvent args)
+        {
+            IngestionBlockerComponent blocker;
+            if (_inventorySystem.TryGetSlotEntity(args.Uid, "head", out var headUid) &&
```

Comment:
Check if it's not already cancelled first because getting slot entities is expensive rn.

Reply:
ok